src/main/java/processor/Storage.java

```diff
@@ -0,0 +1,140 @@
+     * creates a new file for input.
+     */
+    public static void init() {
+        try {
+            File taskFile = new File("data/tasks.txt");
+            if (!taskFile.exists()) {
+                taskFile.getParentFile().mkdirs();
+                taskFile.createNewFile();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Attempts to save tasks to the created .txt file on the local computer. Tasks are first packaged before storage.
+     *
+     * @param processor Duke's processor.
+     * @throws IOException Throws an exception for errors pertaining to saving the tasks in the external text file.
+     */
+    public static void saveTasks(DukeProcessor processor) throws IOException {
+        FileWriter fw = new FileWriter("data/tasks.txt");
+        fw.write("");
+        for (int i = 0; i < processor.getTaskList().size(); i++) {
+            Task task = processor.getTaskList().get(i);
+            String taskStorageString = packageTask(task);
+            fw.append(taskStorageString);
+
+            if (i != processor.getTaskList().size() - 1) {
+                fw.append(System.getProperty("line.separator"));
+            }
+        }
+
+        fw.close();
+    }
+
+    /**
+     * Attempts to load tasks from the .txt file on the local computer, and processes packaged tasks from the file
+     * before returning the list of saved tasks.
+     *
+     * @param processor Duke's processor.
+     * @return The list of tasks saved previously by Duke.
+     * @throws IOException Throws an exception for errors pertaining to loading the tasks in the external text file.
+     * @throws DukeException Throws an exception if the method processPackagedTask fails.
+     */
+    public static List<Task> loadTasks(DukeProcessor processor) throws IOException, DukeException {
+        File taskFile = new File("data/tasks.txt");
+
+        if (!taskFile.exists()) {
+            throw new IOException("File not found");
+        }
+
+        Scanner sc = new Scanner(taskFile);
+
+        List<Task> outputTaskList = new ArrayList<Task>();
+
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            Task task = processPackagedTask(line);
+            outputTaskList.add(task);
+        }
+
+        return outputTaskList;
+    }
+
+    /**
+     * Packages the tasks into a format that can be stored and re-processed when loaded in the future.
+     *
+     * @param task Task to be packaged.
+     * @return Task object in the form of a packaged String, ready to be stored.
+     */
+    private static String packageTask(Task task) {
+        String output = "";
+        int doneIndicator = 0;
+
+        if (task.isDone()) {
+            doneIndicator = 1;
+        }
+
+        if (task instanceof TodoTask) {
+            output = String.format("%s^_^%d^_^%s", "TODO", doneIndicator, task.getDescription());
```

Comment:
you can probably SLAP this out? string formatting is a lower-level operation than what is happening here.